src/python/pants/backend/python/rules/run_setup_py.py

```diff
@@ -525,9 +519,11 @@ async def get_ancestor_init_py(
     )
     # Find the ancestors of all dirs containing .py files, including those dirs themselves.
     source_dir_ancestors: Set[Tuple[str, str]] = set()  # Items are (src_root, path incl. src_root).
-    for fp in sources.snapshot.files:
-        source_dir_ancestor = os.path.dirname(fp)
-        source_root = source_roots.strict_find_by_path(fp).path
+    source_roots = await MultiGet(
+        Get[SourceRoot](SourceRootRequest(path)) for path in sources.snapshot.files
```

Comment:
You can use `sources.files` for the same thing. This shortcut didn't exist when this rule was first written.